### ISSUE TITLE
Can't Install Overwatch. "A 64-bit operating system is required."

### ISSUE THREAD

**Issue by RekkyRek** [gamax92/wine-overwatch#5]
## My system information
OS: elementary OS 0.4 Loki
Wine: "wine-2.8 (Staging)" (from make install)
CPU: Intel® Core™ i7-4790K CPU @ 4.00GHz × 8
RAM: 16GB
GPU: Asus Strix GTX 1080

## Wine Console Log
[Pastebin of Log](https://pastebin.com/W01jN7Jt)

## Issue walkthrough
I followed the instructions on how to install wine-overwatch then tried installing Overwatch via the Overwatch installer you can download on Blizzards website. I changed wine to XP then I ran `wine Overwatch-Setup.exe` and i got the installer to pop up.

![](https://i.imgur.com/iwABGqV.png)

It did some initializing.

![](https://i.imgur.com/NtWmGlP.png)

Then the actual installer popped up.

![](https://i.imgur.com/vD3DbQb.png)

So far so good. After pressing continue it installed the Blizzard App and an odd looking login window popped up.

![](https://i.imgur.com/KNw7HCu.png)

The text under the close button says 

> Your browser's cookies are disabled. Please reenable cookies to continue.

I ignored this text since i don't know any way of enabling them in that window. Then i logged in i got this message: 

![](https://i.imgur.com/asYH261.png)

Which was quite logical considering the previous quote, so i clicked Continue and got a message saying 

> A 64-bit operating system is required.

![](https://i.imgur.com/SSKRx6V.png)

This was where i got stuck. It threw me into the Blizzard App which obviously had some stuff missing. 

![](https://i.imgur.com/X7xaHRF.jpg)

**Comment by gamax92** [comment]
What instructions did you use for building wine?

**Comment by RekkyRek** [comment]
@gamax92 [https://github.com/gamax92/wine-overwatch/blob/overwatch-2.7/README](https://github.com/gamax92/wine-overwatch/blob/overwatch-2.7/README)

**Comment by zekesonxx** [comment]
That builds a 32 bit wine. You need to build a multiarch wine.
https://wiki.winehq.org/Building_Wine#Shared_WoW64

**Comment by RekkyRek** [comment]
@zekesonxx alright. Will try that tomorrow 👍

**Comment by alexandrebouthinon** [comment]
@zekesonxx Hi, same issue for me. I compile Wine using the doc https://github.com/gamax92/wine-overwatch/blob/overwatch-2.7/README but I've a question...
Will the installation script work ? I mean the "install" option in the Makefile

**Comment by zekesonxx** [comment]
@alexandrebouthinon again, the install instructions in the README build a 32-bit Wine, not the multiarch Wine you need.

https://wiki.winehq.org/Building_Wine#Shared_WoW64

Instructions:

1. Download the repo somewhere. I'll refer to it as `$wineoverwatchsrc`
2. Make three directories somewhere, which I'll refer to as `$wine32-build`, `$wine64-build`, and `$wine-install`
3. I'd recommend building with `-mtune=native -O3 -pipe -fstack-protector-strong`, to irk as much performance out of the largely CPU-bound experience as you can:
    * `export CFLAGS="-mtune=native -O3 -pipe -fstack-protector-strong"`
    * `export CPPFLAGS="-mtune=native -O3 -pipe -fstack-protector-strong"`
5. `cd $wine64-build`
6. `$wineoverwatchsrc/configure --enable-win64`
7. `make -jX`, where X is the number of CPU cores you want to compile with
8. `cd $wine32-build`
9. `PKG_CONFIG_PATH=/usr/lib32 $wineoverwatchsrc/configure --with-wine64=$wine64-build`
10. `make -jX`, same deal as above.
11. `make DESTDIR=$wine-install install-lib`.  
You need `$wine-install` to be an absolute path, so something like `/home/zekesonxx/bullshit/wo27/wine-install` not something like `../wine-install`.
12. `cd $wine64-build`
13. `make DESTDIR=$wine-install install-lib`. Again, `$wine-install` needs to be absolute.


TL;DR: make three directories to work in, build 64 bit, build 32 bit, install 32 bit, install 64 bit, and go from there.

**Comment by alexandrebouthinon** [comment]
@zekesonxx Thanks for this tutorial. I forgot the third directory. I'll try this today :)

**Comment by gamax92** [comment]
Ack. I'll change that soon, wine's default README is fairly misleading for Overwatch.

**Comment by gamax92** [comment]
@RekkyRek @alexandrebouthinon Have you two been able to build this repository correctly? Would like to close this issue.

**Comment by RekkyRek** [comment]
@gamax92 nope. I just gave up and dual booted windows :/  If anyone would like to make a video explaining in detail how to build it and install overwatch correctly, that would be great. I found a tutorial for building it but not installing it, when i tried to it didn't work, same as in original post :/

**Comment by zekesonxx** [comment]
@RekkyRek Lutris offers a [precompiled version](https://lutris.net/games/overwatch) of wine-ow, if you're having issues compiling.

Doing a multilib compile on Ubuntu (and friends) seems to be needlessly complicated.

**Comment by r-richmond** [comment]
So following @zekesonxx 's tutorial I got to step 12 before receiving 

macos 10.12.5 if relevant
ran these before following the guide
```
brew cask install xquartz
brew install --only-dependencies --devel wine
```

```
STRIPPROG="strip" ../wine-overwatch/tools/install-sh  dlls/npptools/npptools.dll.so /Applications/Wine/wine-install//usr/local/lib64/wine/npptools.dll.so
../wine-overwatch/tools/install-sh -m 644  dlls/npptools/npptools.dll.fake /Applications/Wine/wine-install//usr/local/lib64/wine/fakedlls/npptools.dll
gcc -m64 -c -o signal_x86_64.o ../../../wine-overwatch/dlls/ntdll/signal_x86_64.c -I. \
  -I../../../wine-overwatch/dlls/ntdll -I../../include -I../../../wine-overwatch/include \
  -D__WINESRC__ -D_NTSYSTEM_ -D_REENTRANT -fPIC -Wall -pipe -fno-strict-aliasing \
  -Wdeclaration-after-statement -Wempty-body -Wignored-qualifiers -Wstrict-prototypes -Wtype-limits \
  -Wvla -Wwrite-strings -Wpointer-arith -gdwarf-2 -gstrict-dwarf -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=0
../../../wine-overwatch/dlls/ntdll/signal_x86_64.c:1222:45: warning: taking the absolute value of unsigned type 'ULONG_PTR' (aka 'unsigned long') has no effect [-Wabsolute-value]
        case DW_OP_abs:         stack[sp] = labs(stack[sp]); break;
                                            ^
../../../wine-overwatch/dlls/ntdll/signal_x86_64.c:1222:45: note: remove the call to 'labs' since unsigned values cannot be negative
        case DW_OP_abs:         stack[sp] = labs(stack[sp]); break;
                                            ^~~~
../../../wine-overwatch/dlls/ntdll/signal_x86_64.c:2711:10: warning: overflow converting case value to switch condition type (-1 to 65535) [-Wswitch]
    case TRAP_x86_UNKNOWN:   /* Unknown fault code */
         ^
../../../wine-overwatch/dlls/ntdll/signal_x86_64.c:2759:10: error: use of undeclared identifier 'SI_KERNEL'
    case SI_KERNEL:    /* Software breakpoint */
         ^
../../../wine-overwatch/dlls/ntdll/signal_x86_64.c:2767:33: error: use of undeclared identifier 'SI_KERNEL'
        if (siginfo->si_code == SI_KERNEL && TRAP_sig(ucontext) != TRAP_x86_BPTFLT)
                                ^
../../../wine-overwatch/dlls/ntdll/signal_x86_64.c:2768:77: warning: format specifies type 'long long' but the argument has type '__uint16_t' (aka 'unsigned short') [-Wformat]
            FIXME( "si_code == SI_KERNEL, but TRAP_sig(context) == %lld\n", TRAP_sig(ucontext) );
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
                                                                   %hu
../../../wine-overwatch/dlls/ntdll/signal_x86_64.c:270:30: note: expanded from macro 'TRAP_sig'
#define TRAP_sig(context)    ((context)->uc_mcontext->__es.__trapno)
                             ^
../../../wine-overwatch/include/wine/debug.h:91:49: note: expanded from macro '__WINE_DBG_LOG'
    wine_dbg_log( __dbcl, __dbch, __FUNCTION__, args); } } while(0)
                                                ^~~~
3 warnings and 2 errors generated.
make[1]: *** [signal_x86_64.o] Error 1
make: *** [dlls/ntdll] Error 2
```

question for @gamax92 . Is it possible to get the official 
`brew install wine` command to install from this repo?
I looked up various commands like `brew install wine head` or `brew install wine --devel` but those appear to be hardcoded  [here](https://github.com/Homebrew/homebrew-core/blob/master/Formula/wine.rb) 

note: I'm definitely in a little bit over my head here but any help is greatly appreciated

**Comment by zekesonxx** [comment]
Closing and locking before this gets derailed again. If you're getting this error you fucked up at doing a multilib Wine build. See previous discussion and the wiki for how to properly do one.

If you're getting a different error don't post here.
